src/analyzer/protocol/mysql/mysql-protocol.pac

```diff
@@ -240,13 +240,14 @@ type Handshake_v10 = record {
 	status_flags           : uint16;
 	capability_flags_2     : uint16;
 	auth_plugin_data_len   : uint8;
-	reserved               : padding[10]; 
-	auth_plugin_data_part_2: bytestring &length=13;
+	reserved               : padding[10];
+	auth_plugin_data_part_2: bytestring &length=auth_plugin_data_part_2_len;
 	have_plugin : case ( ( capability_flags_2 << 16 ) & CLIENT_PLUGIN_AUTH ) of {
 		CLIENT_PLUGIN_AUTH -> auth_plugin: NUL_String;
 		0x0 -> none    : empty;
 	};
 } &let {
+	auth_plugin_data_part_2_len = (auth_plugin_data_len > 8 && (auth_plugin_data_len - 8) > 13) ? auth_plugin_data_len - 8 : 13;
```

Comment:
Wouldn't this be easier as `auth_plugin_data_len > 21` and a comment for the magic number?